### ISSUE TITLE
TensorBoard callback should not close and reopen writers to enable reuse

### ISSUE THREAD

**Issue by bersbersbers** [tensorflow/tensorflow#34449]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes, see below
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): OpenSUSE
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.0.0-rc2-26-g64c3d38 2.0.0
- Python version: 3.7.5

**Describe the current behavior**
Currently, the `tf.keras.callbacks.TensorBoard.set_model` function closes the `train` writer, only to reopen it two lines later:

https://github.com/tensorflow/tensorflow/blob/7eef97bba7a9018d6b0630418d0810f94c1d2222/tensorflow/python/keras/callbacks.py#L1535-L1537

This complicates sharing the writer with other callbacks, e.g., in my code below: this code throws a `RuntimeError: SummaryWriter is already closed`.

**Describe the expected behavior**
`tf.keras.callbacks.TensorBoard.set_model` should only close those writers that will not be needed any longer.

**Code to reproduce the issue**
```
# import keras
import numpy as np
import tensorboard.plugins.hparams.api as hp
import tensorflow.keras as keras

tb_callback = keras.callbacks.TensorBoard(log_dir='/tmp')
tb_train_writer = tb_callback._get_writer(tb_callback._train_run_name)  # pylint:disable=protected-access
hp_callback = hp.KerasCallback(tb_train_writer, {'param': 42})
model = keras.applications.VGG16()
model.compile(optimizer=keras.optimizers.Adam(), loss='categorical_crossentropy')
model.fit(np.empty((1, 224, 224, 3)), np.empty((1, 1000)), callbacks=[tb_callback, hp_callback])
```

**Comment by oanush** [comment]
Issue replicating for version TF-2.0. Kindly find the [gist](https://colab.sandbox.google.com/gist/oanush/f64e85cedff9e96dfe12b36ef3fba472/34449.ipynb) of colab.Thanks!

**Comment by bersbersbers** [comment]
This can be closed: TF 2.2 does not even create writers any more until `set_model`.

**Comment by jackshi0912** [comment]
What is the accepted API for using a shared file_writer with tensorboard callback?
